pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go

```diff
@@ -278,11 +281,33 @@ type addColumnSpec struct {
 	onUpdate *scpb.ColumnOnUpdateExpression
 	comment  *scpb.ColumnComment
 	unique   bool
+	notNull  bool
 }
 
-// addColumn is a helper function which adds column element targets and ensures
-// that the new column is backed by a primary index, which it returns.
+// addColumn adds a column as specified in the `spec`. It delegates most of the work
+// to addColumnIgnoringNotNull and handles NOT NULL constraint itself.
+// It contains version gates to help ensure compatibility in mixed version state.
+// Read comments in `ColumnType` message in `elements.proto` for details.
 func addColumn(b BuildCtx, spec addColumnSpec, n tree.NodeFormatter) (backing *scpb.PrimaryIndex) {
+	backing = addColumnIgnoringNotNull(b, spec, n)
+	if spec.notNull {
+		cnne := scpb.ColumnNotNull{
+			TableID:  spec.tbl.TableID,
+			ColumnID: spec.col.ColumnID,
+		}
+		if backing != nil {
+			cnne.IndexIDForValidation = backing.IndexID
+		}
+		b.Add(&cnne)
+	}
+	return backing
+}
+
+// addColumnIgnoringNotNull is a helper function which adds column element targets and ensures
+// that the new column is backed by a primary index, which it returns.
+func addColumnIgnoringNotNull(
```

Comment:
Does anybody call `addColumnIgnoringNotNull` besides `addColumn`? I didn't see any callers. Given that, it feels like this should just be logic in `addColumn`.

Reply:
done!